embedding-stores/pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/runtime/PgVectorEmbeddingStoreRecorder.java

```diff
@@ -19,10 +18,13 @@ public class PgVectorEmbeddingStoreRecorder {
     public Function<SyntheticCreationalContext<PgVectorEmbeddingStore>, PgVectorEmbeddingStore> embeddingStoreFunction(
             PgVectorEmbeddingStoreConfig config, String datasourceName) {
         return context -> {
-            AgroalDataSource dataSource = Optional.ofNullable(datasourceName)
-                    .map(DataSourceLiteral::new)
-                    .map(dl -> context.getInjectedReference(AgroalDataSource.class, dl))
-                    .orElse(context.getInjectedReference(AgroalDataSource.class, new Default.Literal()));
```

Comment:
This was a sneaky one... The problem here was that `context.getInjectedReference(AgroalDataSource.class, new Default.Literal())` is not wrapped a lambda expression, therefore it gets evaluated always...

Reply:
Ouch...

Reply:
One more reason to hate lambdas  @edeandrea

Reply:
Thanks for the zing @geoand :) I'll still keep my stance that, generally speaking, functional code is still easier to read and figure out what's going on than imperative code. 

That being said, in this particular case, one has to understand the subtle differences between `Optional.orElse` and `Optional.orElseGet`.